backend/pkg/api/data_access/notifications.go

```diff
@@ -77,11 +100,534 @@ func (d *DataAccessService) UpdateNotificationSettingsClients(ctx context.Contex
+	// 	if err != nil {
+	// 		return fmt.Errorf(`error retrieving data for validator dashboard notifications: %w`, err)
+	// 	}
+
+	// 	return nil
+	// })
+
+	err = wg.Wait()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error retrieving dashboard notification data: %w", err)
+	}
+
+	// -------------------------------------
+	// Evaluate the data
+	type NotificationSettingsDashboardsInfo struct {
+		IsAccountDashboard bool // if false it's a validator dashboard
+		DashboardId        uint64
+		DashboardName      string
+		GroupId            uint64
+		GroupName          string
+		// if it's a validator dashboard, Settings is NotificationSettingsAccountDashboard, otherwise NotificationSettingsValidatorDashboard
+		Settings interface{}
+		ChainIds []uint64
+	}
+	settingsMap := make(map[string]*NotificationSettingsDashboardsInfo)
+
+	for _, event := range events {
+		eventSplit := strings.Split(event.Filter, ":")
+		if len(eventSplit) != 3 {
+			continue
+		}
+		dashboardType := eventSplit[0]
+
+		if _, ok := settingsMap[event.Filter]; !ok {
+			if dashboardType == ValidatorDashboardEventPrefix {
+				settingsMap[event.Filter] = &NotificationSettingsDashboardsInfo{
+					Settings: t.NotificationSettingsValidatorDashboard{
+						GroupOfflineThreshold:  GroupOfflineThresholdDefault,
+						MaxCollateralThreshold: MaxCollateralThresholdDefault,
+						MinCollateralThreshold: MinCollateralThresholdDefault,
+					},
+				}
+			} else if dashboardType == AccountDashboardEventPrefix {
+				settingsMap[event.Filter] = &NotificationSettingsDashboardsInfo{
+					Settings: t.NotificationSettingsAccountDashboard{
+						ERC20TokenTransfersValueThreshold: ERC20TokenTransfersValueThresholdDefault,
+					},
+				}
+			}
+		}
+
+		switch settings := settingsMap[event.Filter].Settings.(type) {
+		case t.NotificationSettingsValidatorDashboard:
+			switch event.Name {
+			case types.ValidatorIsOfflineEventName:
+				settings.IsValidatorOfflineSubscribed = true
+			case types.GroupIsOfflineEventName:
+				settings.IsGroupOfflineSubscribed = true
+				settings.GroupOfflineThreshold = event.Threshold
+			case types.ValidatorMissedAttestationEventName:
+				settings.IsAttestationsMissedSubscribed = true
+			case types.ValidatorProposalEventName:
+				settings.IsBlockProposalSubscribed = true
+			case types.ValidatorUpcomingProposalEventName:
+				settings.IsUpcomingBlockProposalSubscribed = true
+			case types.SyncCommitteeSoon:
+				settings.IsSyncSubscribed = true
+			case types.ValidatorReceivedWithdrawalEventName:
+				settings.IsWithdrawalProcessedSubscribed = true
+			case types.ValidatorGotSlashedEventName:
+				settings.IsSlashedSubscribed = true
+			case types.RocketpoolCollateralMinReached:
+				settings.IsMinCollateralSubscribed = true
+				settings.MinCollateralThreshold = event.Threshold
+			case types.RocketpoolCollateralMaxReached:
+				settings.IsMaxCollateralSubscribed = true
+				settings.MaxCollateralThreshold = event.Threshold
```

Comment:
I'm not quite sure if I understand correctly - the goal is to retrieve the current status of user subscriptions, right? Why are we setting everything to true here, I would expect each value to be set according to the value saved in the DB. Is this just default values? Am I missing something?